### ISSUE TITLE
Ferret crashes when directly comparing a variable attribute (e.g. units) to a string

### ISSUE THREAD

**Issue by karlmsmith** [NOAA-PMEL/Ferret#1492]
**Reported by andrew wittenberg on 4 Dec 2014 20:36 UTC**
Defining an intermediate variable is a workaround, but the crash should be avoided.

```
 	NOAA/PMEL TMAP
 	FERRET v6.93  
 	Linux 2.6.32-504.el6.x86_64 64-bit - 11/13/14
 	 4-Dec-14 15:31     

yes? let/unit="m" a = 1
yes? let b = a.units
yes? list b eq "m"        !this works
             VARIABLE : B EQ "m"
             X        : 1
          1.000
yes? list a.units eq "m"  !but this crashes Ferret
STOP  bad op err
```

Migrated-From: http://dunkel.pmel.noaa.gov/trac/ferret/ticket/2220

**Comment by karlmsmith** [comment]
**Comment by andrew.wittenberg on 4 Dec 2014 20:49 UTC**
Here is a related crash:

```
 	NOAA/PMEL TMAP
 	FERRET v6.93  
 	Linux 2.6.32-504.el6.x86_64 64-bit - 11/13/14
 	 4-Dec-14 15:42     

yes? let/unit="m" a = 1
yes? list is_element_of_str(a.units,{"m","a"})
STOP gcf_get_axis_srcs
```

And another one -- which is worse, in that even the intermediate variable doesn't work:

```
 	NOAA/PMEL TMAP
 	FERRET v6.93  
 	Linux 2.6.32-504.el6.x86_64 64-bit - 11/13/14
 	 4-Dec-14 15:41     

yes? let/unit="m" a = 1
yes? let b = a.units
yes? list is_element_of_str(b,{"m","a"})
**ERROR Ferret crash; signal = 11
```

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 4 Dec 2014 22:20 UTC**
There seems to be a difference in behavior with user-defined variables vs variables from datsets, where defining an intermediate variable works, but not the expression var.att:



```
yes? use levitus_climatology
yes? sh att temp
     attributes for dataset: /home/users/tmap/ferret/linux/fer_dsets/data/levitus_climatology.cdf
 TEMP.missing_value = -1.E+10
 TEMP._FillValue = -1.E+10
 TEMP.long_name = TEMPERATURE 
 TEMP.history = From levitus_climatology 
 TEMP.units = DEG C 

yes? let a = temp.units
yes? list is_element_of_str (a,{"Kelvin", "Celsius", "DEG C"})
             VARIABLE : IS_ELEMENT_OF_STR (A,{"Kelvin", "Celsius", "DEG C"})
             FILENAME : levitus_climatology.cdf
             FILEPATH : /home/users/tmap/ferret/linux/fer_dsets/data/
             X        : 1
          1.000

yes? list is_element_of_str (temp.long_name,{"tmp", "temp", "TEMPERATURE"})
STOP gcf_get_axis_srcs
```

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 9 Dec 2014 18:45 UTC**
When an expression that is a varname.attname is sent to a grid-changing function (which many of the string functions are), the variable's grid was not being computed. 

This is fixed in init_uvar, is_do_gc_fcn, is_uvar_grid.

**Comment by karlmsmith** [comment]
**Comment by andrew.wittenberg on 11 Dec 2014 17:11 UTC**
Almost there... the examples in my comment:1 are fixed.  But the original example still gives errors (though now thankfully not a crash):


```
 	NOAA/PMEL TMAP
 	FERRET v6.931 (beta)
 	Linux 2.6.32-504.1.3.el6.x86_64 64-bit - 12/10/14
 	11-Dec-14 12:08     

yes? let/unit="m" a = 1
yes? let b = a.units
yes? list b eq "m"
             VARIABLE : B EQ "m"
             X        : 1
          1.000
yes? list a.units eq "m"  !but gives errors
 **ERROR: attribute undefined or invalid attribute name: A.UNITS EQ "m"
          attribute UNITS EQ "m" not found for variable a
 **ERROR: variable unknown or not in data set: A.UNITS EQ "m"
```

Seems like a parsing problem following the period (".").

Downgrading the priority a bit, since this no longer crashes Ferret.

**Comment by karlmsmith** [comment]
**Comment by andrew.wittenberg on 27 Jan 2017 19:12 UTC**
This seems to be fixed in v7.031b (dated 1/26/17).
